rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -0,0 +1,357 @@
+
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <cstdio>
+#if !ENABLE_HIP
+#include <CL/cl.h>
+#endif
+#include <stdexcept>
+#include <vx_ext_amd.h>
+#include <cstring>
+#include "commons.h"
+#include "tensor.h"
+
+vx_enum vx_mem_type(RocalMemType mem)
+{
+    switch(mem)
+    {
+        case RocalMemType::OCL:
+        {
+            return VX_MEMORY_TYPE_OPENCL;
+        }
+        break;
+        case RocalMemType::HOST:
+        {
+            return VX_MEMORY_TYPE_HOST;
+        }
+        break;
+        case RocalMemType::HIP:
+        {
+            return VX_MEMORY_TYPE_HIP;
+        }
+        break;
+        default:
+            throw std::runtime_error("Memory type not valid");
+    }
+}
+
+vx_size tensor_data_size(RocalTensorDataType data_type)
+{
+    switch(data_type)
+    {
+        case RocalTensorDataType::FP32:
+        {
+            return sizeof(vx_float32);
+        }
+        break;
+        case RocalTensorDataType::FP16:
+        {
+            return sizeof(vx_int16);
```

Comment:
this will get changed to vx_float16. There is another PR to fix that in amd OpenVX